### ISSUE TITLE
Mockito cannot mock/spy final class

### ISSUE THREAD

**Issue by vicpinm** [tmurakami/dexopener#15]
Hi, I have a question related with your library. Can I mock classes with are under other modules of my project? For example, projects usually have three modules: view, domain and data. When I run my test, it says that one class from my data module cannot be mocked.  I think is related with this:

"The final classes/methods you can mock are only those under the package of your app's BuildConfig. Therefore, you cannot mock final classes/methods you don't own, such as Android system classes and third-party libraries."

But in my case, I do own that classes, but they are in other modules in my project. Is there any way to workaround this?

**Comment by tmurakami** [comment]
If you want to test `view` module, could you try the following steps?

1. Create a dummy BuildConfig in your `view/src/androidTest` directory.

```kotlin
// Kotlin
package com.example.yourapp // root package of three modules

object BuildConfig {
    const val APPLICATION_ID = "com.example.yourapp.view" // the applicationId of `view` module
}
```

2. Create custom AndroidJUnitRunner in your `view/src/androidTest` directory.

```kotlin
// Kotlin
package com.example.yourapp.view.test

class CustomJUnitRunner : AndroidJUnitRunner() {
    override fun newApplication(cl: ClassLoader, className: String?, context: Context): Application {
        DexOpener.builder(context)
            .buildConfig(com.example.yourapp.BuildConfig::class.java) // Set the BuildConfig created at #1
            .build()
            .installTo(cl)
        return super.newApplication(cl, className, context)
    }
}
```

3. Specify CustomJUnitRunner as the default test runner in your `view/build.gradle`.

```groovy
android {
    defaultConfig {
        testInstrumentationRunner 'com.example.yourapp.view.test.CustomJUnitRunner'
    }
}
```

**Comment by JorgeCastilloPrz** [comment]
And what would you do if the class to mock is defined on another module, and the module is pure kotlin / java? (no `android {}` block on build.gradle)

**Comment by tmurakami** [comment]
I have just released version 2.0.0-alpha01 which fixed this issue. You can now manually specify your root package. Please see the README and the examples for more information.
